.gitignore

```diff
@@ -1,50 +1,16 @@
-*.o
-*.Po
-*.so*
-*~
-*.a
-*.la
-*.lo
-*.Tpo
-*.swp
-*.bin
-*.map
-*.ri
-*.orig
-*.rej
-*.rom
-*.man
-.version
-*.x
-.build
-*.dis
-*.lst
-*.log
-*.html
-*.js
-*.png
-*.md5
-*.xml
-*.css
-*.xsd
-*.patch
-*.xslt
-*.mbox
+# Do not mass copy/paste/diverge in every project your non-project
+# specific, personal git ignore preferences like: *.mbox, *.patch, *.rej,
+# cscope*, TAGS, *~,...
+#
+# Instead, set your non-project specific, personal git ignore preference
+# only once for all git projects in your home directory:
+#     https://git-scm.com/docs/gitignore
+# So this does not interfere other people's "git status" preferences.
+#
+# Also, don't use `git add *` or `git commit -a` and git push blindly
+# without a look at which files are getting pushed. This is bad no
+# matter what is in git ignore.
```

Comment:
Hmm, not sure about this. Is there any harm to the overblown gitignore file? 

I mean we frequently have contributors not so used to open-source tools, so misuse of "git commit -a" is a non-zero risk and large gitignore may provide some protection against this (and bring value to the project).

Reply:
> Is there any harm to the overblown gitignore file?

Yes :-)

> may provide "some protection" against this

"Some" protection is worse than none.

The only way for _both_ authors and reviewers to miss spurious files is to NOT review. "Some" protection teaches contributors and reviewers that not reviewing is fine and that `git commit -a` is fine - because it works most of the time thanks to "some" protection! Until a file not "protected" by .gitignore gets added and missed.

Moreover, even if some people "know better" and really want to use `git commit -a` then it's much "better" even _for them_ to configure it once for all in the $HOME/gitignore file and avoid an "inconsistent git commit -a experience" across repos.